plugins/inputs/dnac/dnac.go

```diff
@@ -0,0 +1,170 @@
+package dnac
+
+// dnac.go
+
+import (
+	"net/url"
+
+	dnac_sdk "github.com/cisco-en-programmability/dnacenter-go-sdk/v3/sdk"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Dnac struct {
+	DnacBaseURL   string
+	Username      string
+	Password      string
+	SSLVerify     string
+	Debug         string
+	ClientHealth  bool
+	NetworkHealth bool
+	Log           telegraf.Logger
+	Client        *dnac_sdk.Client
+}
+
+func (d *Dnac) Description() string {
+	return "a Cisco DNAC plugin"
+}
+
+func (d *Dnac) SampleConfig() string {
+	return `
+  ## Specify DNAC Base URL
+  dnacbaseurl = "https://sandboxdnac.cisco.com"
+  ## Specify Credentials
+  username = "devnetuser"
+  password = "Cisco123!"
+  ## Debug true/false
+  debug = "false"
+  ## SSL Verify true/false
+  sslverify = "false"
+  ## Report Client Health
+  clienthealth = true
+  ## Report Network Health
+  networkhealth = true
+`
+}
+
+// Init is for setup, and validating config.
+func (d *Dnac) Init() error {
+	return nil
+}
```

Comment:
In order to get the test to work with the httpmock, I had to separate the client initialization from the authentication. Otherwise when I init the client in the test it fails to connect before I can activate the mock. Happy to have some input here as to a better way to tackle that.

Reply:
Ok, I see the Init function isn't called when I instantiate it from the test. I added the InitClient and Auth functions to the Init method.